cookbook/routing/redirect_trailing_slash.rst

```diff
@@ -37,13 +37,38 @@ system, as explained below:
 
 .. configuration-block::
 
+    .. code-block:: php-annotations
+
+        // src/AppBundle/Controller/RedirectingController.php
+        namespace AppBundle\Controller;
+
+        use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+        use Symfony\Component\HttpFoundation\Request;
+
+        class RedirectingController extends Controller
+        {
+            /**
+             * @Route("/{url}", name="remove_trailing_slash",
+             *     requirements={"url" = ".*\/$"}, methods={"GET"})
+             */
+            public function removeTrailingSlashAction(Request $request)
+            {
+                $pathInfo = $request->getPathInfo();
+                $requestUri = $request->getRequestUri();
+
+                $url = str_replace($pathInfo, rtrim($pathInfo, ' /'), $requestUri);
+
+                return $this->redirect($url, 301);
```

Comment:
We already have the complete code above the configurations so I would reduce this to the bare minimum like this:

```rst
.. configuration-block::

    .. code-block:: php+annotations::

        // src/AppBundle/Controller/RedirectingController.php
        
        // ...
        class RedirectingController extends Controller
        {
            /**
             * @Route("/{url}", name="remove_trailing_slash",
             *     requirements={"url" = ".*\/$"}, methods={"GET"})
             */
            public function removeTrailingSlashAction(Request $request)
            {
                // ...
            }
        }
```